src/test/mir-opt/unreachable.rs

```diff
@@ -0,0 +1,32 @@
+use std::mem::transmute;
+
+enum Empty {}
+
+fn main() {
+    unsafe {
+        let _x: Empty = transmute(());
```

Comment:
The other thing you could do is write the test like this so that you can still see if your optimization kicks in but it won't actually do any UB at runtime so the executable will run successfully:

```rust
fn empty() -> Option<Empty> {
    None
}

fn main() {
    if let Some(_x) = empty() {
        let mut _y;

        if true {
            _y = 21;
        } else {
            _y = 42;
        }

        match _x { }
    }
}
```